src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -216,7 +216,7 @@ internal BoundExpression ConvertPatternExpression(TypeSymbol inputType, CSharpSy
                     case ConversionKind.Identity:
                     case ConversionKind.ImplicitReference:
                     case ConversionKind.Unboxing:
-                    case ConversionKind.NullLiteral:
+                    case ConversionKind.DefaultOrNullLiteral:
```

Comment:
Apparently this line affects no tests whatsoever. I'll see if I can hit it, or I'll file an issue.